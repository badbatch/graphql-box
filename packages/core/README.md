# @graphql-box/core

The GraphQL Box core module.

[![npm version](https://badge.fury.io/js/%40graphql-box%2Fcore.svg)](https://badge.fury.io/js/%40graphql-box%2Fcore)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](LICENSE)

## Installation

```bash
yarn add @graphql-box/core
```

## Usage

Read the GraphQL Box [usage guide](../../README.md#usage).

## Documentation

Read the module API [documentation](docs/README.md).
