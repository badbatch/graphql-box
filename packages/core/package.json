{
  "name": "@graphql-box/core",
  "version": "4.1.6",
  "description": "The GraphQL Box core module.",
  "author": "Dylan Aubrey <dylanaubrey@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/badbatch/graphql-box",
  "bugs": {
    "url": "https://github.com/badbatch/graphql-box/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/badbatch/graphql-box",
    "directory": "packages/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "main": "lib/main/index.js",
  "module": "lib/module/index.mjs",
  "browser": "lib/browser/index.js",
  "types": "lib/types/index.d.ts",
  "keywords": [
    "graphql-box",
    "graphql",
    "client",
    "server",
    "isomorphic"
  ],
  "scripts": {
    "compile": "concurrently npm:compile:*",
    "compile-watch": "concurrently npm:compile-watch:*",
    "compile-watch:main": "yarn run compile:main --watch",
    "compile-watch:types": "yarn run compile:types --watch",
    "compile:browser": "cross-env BABEL_ENV=browser rollup -c ../../rollup.config.js",
    "compile:main": "cross-env BABEL_ENV=main babel ./src --out-dir ./lib/main --extensions \".ts\" --source-maps --config-file ../../babel.config.js",
    "compile:module": "cross-env BABEL_ENV=module babel ./src --out-dir ./lib/module --extensions \".ts\" --out-file-extension \".mjs\" --source-maps --config-file ../../babel.config.js",
    "compile:types": "tsc --build",
    "watch": "cross-env BABEL_ENV=browser rollup -c ../../rollup.config.js -w"
  },
  "dependencies": {
    "serialize-error": "8.1.0"
  },
  "peerDependencies": {
    "@babel/runtime": "< 8",
    "core-js": "< 4",
    "lodash": "< 5"
  }
}