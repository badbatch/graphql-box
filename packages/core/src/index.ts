import * as coreDefs from "./defs";
import Core from "./main";

export {
  Core as default,
  coreDefs,
};
