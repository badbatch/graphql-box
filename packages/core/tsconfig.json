{
  "extends": "../tsconfig.settings.json",
  "compilerOptions": {
    "outDir": "lib/types",
    "rootDir": "src"
  },
  "include": [
    "src"
  ]
}