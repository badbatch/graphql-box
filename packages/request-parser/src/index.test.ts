import { DEFAULT_TYPE_ID_KEY, MUTATION, RequestContext, SUBSCRIPTION } from "@graphql-box/core";
import {
  getRequestContext,
  githubIntrospection,
  requestsAndOptions,
  schemaResolvers,
  schemaTypeDefs,
} from "@graphql-box/test-utils";
import { IntrospectionQuery } from "graphql";
import { makeExecutableSchema } from "graphql-tools";
import { RequestParser, RequestParserDef, UpdateRequestResult } from ".";

describe("@graphql-box/request-parser >>", () => {
  let requestContext: RequestContext;
  let requestParser: RequestParserDef;
  let updatedRequest: UpdateRequestResult;

  describe("query >> no variable >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithoutVariable;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> operation name >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithOperationName;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> variable", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithVariable;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> multiple variables >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithVariables;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> enum variable >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithEnumVariable;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> directive", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithDirective;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> inline fragment >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithInlineFragment;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> inline fragments >> union type >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithUnionInlineFragments;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> fragment spread >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithFragmentSpread;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("query >> fragment option >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.queryWithFragmentOption;
      requestContext = getRequestContext();
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("mutation >> input type variable >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        introspection: githubIntrospection as IntrospectionQuery,
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.nestedInterfaceMutation;
      requestContext = getRequestContext({ operation: MUTATION });
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });

  describe("subscription >>", () => {
    beforeAll(async () => {
      requestParser = await RequestParser.init({
        schema: makeExecutableSchema({ typeDefs: schemaTypeDefs, resolvers: schemaResolvers }),
        typeIDKey: DEFAULT_TYPE_ID_KEY,
      });

      const { options, request } = requestsAndOptions.nestedTypeSubscription;
      requestContext = getRequestContext({ operation: SUBSCRIPTION });
      updatedRequest = await requestParser.updateRequest(request, options, requestContext);
    });

    it("correct request", () => {
      expect(updatedRequest.request).toMatchSnapshot();
    });

    it("correct context data", () => {
      expect(requestContext).toMatchSnapshot();
    });
  });
});
