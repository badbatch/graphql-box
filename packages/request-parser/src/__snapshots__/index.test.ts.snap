// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@graphql-box/request-parser >> mutation >> input type variable >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "mutation.addStar" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "AddStarPayload",
    },
    "mutation.addStar.starrable" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": true,
      "isUnion": false,
      "possibleTypes": Array [
        Object {
          "isEntity": true,
          "typeName": "Repository",
        },
        Object {
          "isEntity": true,
          "typeName": "Topic",
        },
        Object {
          "isEntity": true,
          "typeName": "Gist",
        },
      ],
      "typeIDValue": undefined,
      "typeName": "Starrable",
    },
    "mutation.addStar.starrable.stargazers" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "StargazerConnection",
    },
    "mutation.addStar.starrable.stargazers.edges" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "StargazerEdge",
    },
    "mutation.addStar.starrable.stargazers.edges.node" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "User",
    },
  },
  "operation": "mutation",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> mutation >> input type variable >> correct request 1`] = `
"mutation {
  addStar(input: {clientMutationId: \\"1\\", starrableId: \\"MDEwOlJlcG9zaXRvcnkxMDA0NTUxNDg=\\"}) {
    clientMutationId
    starrable {
      viewerHasStarred
      ... on Repository {
        stargazers(first: 6) {
          edges {
            node {
              name
              login
              id
            }
          }
        }
        id
      }
      __typename
    }
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> directive correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
    "query.organization.repositories" => Object {
      "hasArguments": true,
      "hasDirectives": true,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryConnection",
    },
    "query.organization.repositories.edges" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryEdge",
    },
    "query.organization.repositories.edges.node" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Repository",
    },
    "query.organization.repositories.edges.node.owner" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": true,
      "isUnion": false,
      "possibleTypes": Array [
        Object {
          "isEntity": true,
          "typeName": "Organization",
        },
        Object {
          "isEntity": true,
          "typeName": "User",
        },
      ],
      "typeIDValue": undefined,
      "typeName": "RepositoryOwner",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> directive correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    repositories(first: 6) @include(if: true) {
      edges {
        node {
          description
          homepageUrl
          name
          owner {
            login
            url
            ... on Organization {
              name
              id
            }
            __typename
          }
          id
        }
      }
    }
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> enum variable >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
    "query.organization.repositories" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryConnection",
    },
    "query.organization.repositories.edges" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryEdge",
    },
    "query.organization.repositories.edges.node" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Repository",
    },
    "query.organization.repositories.edges.node.owner" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": true,
      "isUnion": false,
      "possibleTypes": Array [
        Object {
          "isEntity": true,
          "typeName": "Organization",
        },
        Object {
          "isEntity": true,
          "typeName": "User",
        },
      ],
      "typeIDValue": undefined,
      "typeName": "RepositoryOwner",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> enum variable >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    repositories(first: 6, ownerAffiliations: [OWNER, COLLABORATOR]) {
      edges {
        node {
          description
          homepageUrl
          name
          owner {
            login
            url
            ... on Organization {
              name
              id
            }
            __typename
          }
          id
        }
      }
    }
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> fragment option >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> fragment option >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> fragment spread >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> fragment spread >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> inline fragment >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> inline fragment >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> inline fragments >> union type >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.search" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "SearchResultItemConnection",
    },
    "query.search.edges" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "SearchResultItemEdge",
    },
    "query.search.edges.node" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": true,
      "possibleTypes": Array [
        Object {
          "isEntity": true,
          "typeName": "Issue",
        },
        Object {
          "isEntity": true,
          "typeName": "PullRequest",
        },
        Object {
          "isEntity": true,
          "typeName": "Repository",
        },
        Object {
          "isEntity": true,
          "typeName": "User",
        },
        Object {
          "isEntity": true,
          "typeName": "Organization",
        },
        Object {
          "isEntity": true,
          "typeName": "MarketplaceListing",
        },
      ],
      "typeIDValue": undefined,
      "typeName": "SearchResultItem",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> inline fragments >> union type >> correct request 1`] = `
"{
  search(query: \\"react\\", first: 10, type: REPOSITORY) {
    edges {
      node {
        ... on Organization {
          description
          login
          organizationName: name
          id
        }
        ... on Issue {
          bodyText
          number
          title
          id
        }
        ... on MarketplaceListing {
          slug
          shortDescription
          howItWorks
          id
        }
        ... on PullRequest {
          bodyText
          number
          title
          id
        }
        ... on Repository {
          description
          homepageUrl
          name
          id
        }
        __typename
      }
    }
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> multiple variables >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
    "query.organization.repositories" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryConnection",
    },
    "query.organization.repositories.edges" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": false,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "RepositoryEdge",
    },
    "query.organization.repositories.edges.node" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Repository",
    },
    "query.organization.repositories.edges.node.owner" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": true,
      "isUnion": false,
      "possibleTypes": Array [
        Object {
          "isEntity": true,
          "typeName": "Organization",
        },
        Object {
          "isEntity": true,
          "typeName": "User",
        },
      ],
      "typeIDValue": undefined,
      "typeName": "RepositoryOwner",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> multiple variables >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    repositories(first: 6) {
      edges {
        node {
          description
          homepageUrl
          name
          owner {
            login
            url
            ... on Organization {
              name
              id
            }
            __typename
          }
          id
        }
      }
    }
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> no variable >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> no variable >> correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> operation name >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "GetOrganization",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> operation name >> correct request 1`] = `
"query GetOrganization {
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> query >> variable correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "query.organization" => Object {
      "hasArguments": true,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Organization",
    },
  },
  "operation": "query",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> query >> variable correct request 1`] = `
"{
  organization(login: \\"facebook\\") {
    description
    email
    login
    name
    url
    id
  }
}
"
`;

exports[`@graphql-box/request-parser >> subscription >> correct context data 1`] = `
Object {
  "boxID": "123456789",
  "debugManager": null,
  "fieldTypeMap": Map {
    "subscription.emailAdded" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Inbox",
    },
    "subscription.emailAdded.emails" => Object {
      "hasArguments": false,
      "hasDirectives": false,
      "isEntity": true,
      "isInterface": false,
      "isUnion": false,
      "possibleTypes": Array [],
      "typeIDValue": undefined,
      "typeName": "Email",
    },
  },
  "operation": "subscription",
  "operationName": "",
  "queryFiltered": false,
  "request": "",
}
`;

exports[`@graphql-box/request-parser >> subscription >> correct request 1`] = `
"subscription {
  emailAdded {
    emails {
      from
      message
      subject
      unread
      id
    }
    total
    unread
    id
  }
}
"
`;
